lib/logstorage/block_test.go

```diff
@@ -205,3 +207,73 @@ func TestBlockMustInitFromRows_Overflow(t *testing.T) {
 	f(15, 30, 15)
 	f(maxColumnsPerBlock+1000, 1, maxColumnsPerBlock)
 }
+
+func TestBlockUncompressedSizeBytes(t *testing.T) {
+	f := func(timestamps []int64, rows [][]Field) {
+		t.Helper()
+
+		// Build expected JSON and calculate actual serialized size
+		var totalSize int
+		for _, fields := range rows {
+			m := make(map[string]string)
+			m["_time"] = time.RFC3339Nano
+
+			for _, f := range fields {
+				if f.Value == "" {
+					continue // skip empty values
+				}
+				key := f.Name
+				if key == "_msg" || key == "" {
+					key = "_msg" // normalize empty names
+				}
+				m[key] = f.Value
+			}
+
+			b, err := json.Marshal(m)
+			if err != nil {
+				t.Fatalf("failed to marshal JSON: %v", err)
+			}
+			totalSize += len(b) + 1 // +1 for newline if expected
+		}
+
+		b := &block{}
+		b.MustInitFromRows(timestamps, rows)
```

Comment:
Timestamps can be set to arbitrary values here, since they shouldn't affect the `uncompressedSizeBytes()` results.